### PR TITLE
📝 Add documentation for the auth_token

### PR DESCRIPTION
**Pull request by rwalton-arm**
### Description

Add package level docstring describing how to set the authorisation token.

### Test Coverage

- [ ]  This change is covered by existing or additional automated tests.
- [x]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: mbed_targets/__init__.py [hunk 1]
```diff
@@ -1,4 +1,24 @@
-"""Exposes the primary interfaces for the library."""
+"""mbed-targets provides an abstraction layer for boards and platforms supported by Mbed OS.
 
+This package is intended for use by developers of Mbed OS.
+
+To fetch data about private targets, an authentication token must be provided in an
```
Comment (rosiemiller2) [inline mbed_targets/__init__.py]: The user will also need to have an account on os.mbed.com and be member of a vendor team that has permissions to see the private board.

File: mbed_targets/__init__.py [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
+environment variable named `MBED_API_AUTH_TOKEN`. The package also accepts a `.env` file
```
Comment (rosiemiller2) [inline mbed_targets/__init__.py]: I know we tested this and I can't remember what the answer was, which one trumps if both are present?

Comment (rwalton-arm) [reply mbed_targets/__init__.py]: The `.env` file wins as we use `override_system=true` - I'll add clarification on that

File: mbed_targets/__init__.py (hunk 1 above)
Comment (rwalton-arm) [reply mbed_targets/__init__.py]: Good point. I'll add a note about that.

File: mbed_targets/__init__.py [hunk 3]
```diff
@@ -1,4 +1,29 @@
-"""Exposes the primary interfaces for the library."""
+"""mbed-targets provides an abstraction layer for boards and platforms supported by Mbed OS.
 
+This package is intended for use by developers of Mbed OS.
```
Comment (rosiemiller2) [inline mbed_targets/__init__.py]: Sorry one more tiny thing:
```suggestion
This package is intended for use by developers using Mbed OS.
```

**Comment by madchutney** [comment]
Looks good to me.

**Pull request merged**